### PR TITLE
feat: move if-match header validation to ifmatchheadervalidator

### PR DESCRIPTION
**Pull request by jvandaal**

### CODE REVIEW

File: src/AddressRegistry.Api.BackOffice/AddressController-Deregulate.cs [hunk 1]
```diff
@@ -71,19 +72,9 @@ public partial class AddressController
             try
             {
                 // Check if user provided ETag is equal to the current Entity Tag
-                if (ifMatchHeaderValue is not null)
+                if (! await ifMatchHeaderValidator.IsValid(ifMatchHeaderValue, streetNamePersistentLocalId, addressPersistentLocalId, cancellationToken))
```
Comment (lvermeulen) [inline src/AddressRegistry.Api.BackOffice/AddressController-Deregulate.cs:75]: no space between ! and await

File: src/AddressRegistry.Api.BackOffice/AddressController-Regularize.cs [hunk 2]
```diff
@@ -71,19 +72,9 @@ public partial class AddressController
             try
             {
                 // Check if user provided ETag is equal to the current Entity Tag
-                if (ifMatchHeaderValue is not null)
+                if (! await ifMatchHeaderValidator.IsValid(ifMatchHeaderValue, streetNamePersistentLocalId, addressPersistentLocalId, cancellationToken))
```
Comment (lvermeulen) [inline src/AddressRegistry.Api.BackOffice/AddressController-Regularize.cs:75]: no space between ! and await

File: test/AddressRegistry.Tests/BackOffice/Api/WhenApprovingAddress/GivenAddressHasInvalidStatus.cs [hunk 3]
```diff
@@ -22,22 +17,16 @@ namespace AddressRegistry.Tests.BackOffice.Api.WhenApprovingAddress
...
     {
         private readonly AddressController _controller;
         private readonly TestBackOfficeContext _backOfficeContext;
-        private readonly IdempotencyContext _idempotencyContext;
 
         public GivenAddressHasInvalidStatus(ITestOutputHelper testOutputHelper) : base(testOutputHelper)
         {
             _controller = CreateApiBusControllerWithUser<AddressController>();
-            _idempotencyContext = new FakeIdempotencyContextFactory().CreateDbContext();
+            new FakeIdempotencyContextFactory().CreateDbContext();
```
Comment (lvermeulen) [inline test/AddressRegistry.Tests/BackOffice/Api/WhenApprovingAddress/GivenAddressHasInvalidStatus.cs:29]: this is not assigned to anything?

File: test/AddressRegistry.Tests/BackOffice/Api/WhenApprovingAddress/GivenAddressIsRemoved.cs [hunk 4]
```diff
@@ -22,22 +19,16 @@ namespace AddressRegistry.Tests.BackOffice.Api.WhenApprovingAddress
...
     {
         private readonly AddressController _controller;
         private readonly TestBackOfficeContext _backOfficeContext;
-        private readonly IdempotencyContext _idempotencyContext;
 
         public GivenAddressIsRemoved(ITestOutputHelper testOutputHelper) : base(testOutputHelper)
         {
             _controller = CreateApiBusControllerWithUser<AddressController>();
-            _idempotencyContext = new FakeIdempotencyContextFactory().CreateDbContext();
+            new FakeIdempotencyContextFactory().CreateDbContext();
```
Comment (lvermeulen) [inline test/AddressRegistry.Tests/BackOffice/Api/WhenApprovingAddress/GivenAddressIsRemoved.cs:31]: this is not assigned to anything?

File: test/AddressRegistry.Tests/BackOffice/Api/WhenRejectingAddress/GivenAddressHasInvalidStatus.cs [hunk 5]
```diff
@@ -24,12 +22,11 @@ public class GivenAddressHasInvalidStatus : AddressRegistryBackOfficeTest
     {
         private readonly AddressController _controller;
         private readonly TestBackOfficeContext _backOfficeContext;
-        private readonly IdempotencyContext _idempotencyContext;
 
         public GivenAddressHasInvalidStatus(ITestOutputHelper testOutputHelper) : base(testOutputHelper)
         {
             _controller = CreateApiBusControllerWithUser<AddressController>();
-            _idempotencyContext = new FakeIdempotencyContextFactory().CreateDbContext();
+            new FakeIdempotencyContextFactory().CreateDbContext();
```
Comment (lvermeulen) [inline test/AddressRegistry.Tests/BackOffice/Api/WhenRejectingAddress/GivenAddressHasInvalidStatus.cs:29]: this is not assigned to anything?

File: src/AddressRegistry.Api.BackOffice/AddressController-Deregulate.cs (hunk 1 above)
Comment (jvandaal) [reply src/AddressRegistry.Api.BackOffice/AddressController-Deregulate.cs]: Fixed

File: src/AddressRegistry.Api.BackOffice/AddressController-Reject.cs [hunk 6]
```diff
@@ -71,19 +72,9 @@ public partial class AddressController
             try
             {
                 // Check if user provided ETag is equal to the current Entity Tag
-                if (ifMatchHeaderValue is not null)
+                if (! await ifMatchHeaderValidator.IsValid(ifMatchHeaderValue, streetNamePersistentLocalId, addressPersistentLocalId, cancellationToken))
```
[the remark this answers is not in the document]
Comment (jvandaal) [reply src/AddressRegistry.Api.BackOffice/AddressController-Reject.cs]: Fixed

File: test/AddressRegistry.Tests/BackOffice/Api/WhenApprovingAddress/GivenAddressIsRemoved.cs (hunk 4 above)
Comment (jvandaal) [reply test/AddressRegistry.Tests/BackOffice/Api/WhenApprovingAddress/GivenAddressIsRemoved.cs]: Removed

File: test/AddressRegistry.Tests/BackOffice/Api/WhenRejectingAddress/GivenAddressHasInvalidStatus.cs (hunk 5 above)
Comment (jvandaal) [reply test/AddressRegistry.Tests/BackOffice/Api/WhenRejectingAddress/GivenAddressHasInvalidStatus.cs]: Removed

File: test/AddressRegistry.Tests/BackOffice/Api/WhenDeregulatingAddress/GivenAddressHasInvalidStatus.cs [hunk 7]
```diff
@@ -24,12 +22,11 @@ public class GivenAddressHasInvalidStatus : AddressRegistryBackOfficeTest
     {
         private readonly AddressController _controller;
         private readonly TestBackOfficeContext _backOfficeContext;
-        private readonly IdempotencyContext _idempotencyContext;
 
         public GivenAddressHasInvalidStatus(ITestOutputHelper testOutputHelper) : base(testOutputHelper)
         {
             _controller = CreateApiBusControllerWithUser<AddressController>();
-            _idempotencyContext = new FakeIdempotencyContextFactory().CreateDbContext();
+            new FakeIdempotencyContextFactory().CreateDbContext();
```
[the remark this answers is not in the document]
Comment (jvandaal) [reply test/AddressRegistry.Tests/BackOffice/Api/WhenDeregulatingAddress/GivenAddressHasInvalidStatus.cs]: Removed

File: test/AddressRegistry.Tests/BackOffice/Api/WhenApprovingAddress/GivenAddressHasInvalidStatus.cs (hunk 3 above)
Comment (jvandaal) [reply test/AddressRegistry.Tests/BackOffice/Api/WhenApprovingAddress/GivenAddressHasInvalidStatus.cs]: Removed

File: src/AddressRegistry.Api.BackOffice/AddressController-Regularize.cs (hunk 2 above)
Comment (jvandaal) [reply src/AddressRegistry.Api.BackOffice/AddressController-Regularize.cs]: Fixed

**Pull request merged**